WordPress/Classes/Services/NotificationSyncMediator.swift

```diff
@@ -200,6 +212,36 @@ class NotificationSyncMediator {
         updateReadStatus(read, forNoteWithObjectID: notification.objectID)
     }
 
+    private func mark(_ notifications: [Notification], asRead read: Bool = true, completion: ((Error?)-> Void)? = nil) {
+        assert(Thread.isMainThread)
+
+        let noteIDs = notifications.map {
+            $0.notificationId
+        }
+
+        remote.updateReadStatusForNotifications(noteIDs, read: read) { error in
+            if let error = error {
+                DDLogError("Error marking notifications as \(Self.readState(for: read)): \(error)")
+                NotificationSyncMediator()?.invalidateCacheForNotifications(with: noteIDs)
```

Comment:
Since `self` is nil when this line is executed, the existing code called the convenience initializer and instantiated a new instance (`NotificationSyncMediator()`). If it weren't the case, we could inject this as a closure and unit test the invalidation case as well. I was negligent to do this change here as it will require refactoring on a code that existed for some years now.

Reply:
🤔 let's not worry about that right now.